deploy/buildtools/buildconfigfile.py

```diff
@@ -0,0 +1,161 @@
+        acctids_to_sharewith: List of AWS account names to share AGFIs with.
+        hwdb: Object holding all HWDB entries.
+        builds_list: List of build recipe names to build.
+        build_ip_set: List of IPs to use for builds.
+    """
+    args: Namespace
+    agfistoshare: List[str]
+    acctids_to_sharewith: List[str]
+    hwdb: RuntimeHWDB
+    builds_list: List[BuildConfig]
+    build_ip_set: Set[str]
+
+    def __init__(self, args: Namespace) -> None:
+        """
+        Args:
+            args: Object holding arg attributes.
+        """
+        if args.launchtime:
+            launch_time = args.launchtime
+        else:
+            launch_time = strftime("%Y-%m-%d--%H-%M-%S", gmtime())
+
+        self.args = args
+
+        global_build_config_file = None
+        with open(args.buildconfigfile, "r") as yaml_file:
+            global_build_config_file = yaml.safe_load(yaml_file)
+
+        # aws specific options
+        self.agfistoshare = global_build_config_file['agfis-to-share']
+        self.acctids_to_sharewith = global_build_config_file['share-with-accounts'].values()
+
+        # this is a list of actual builds to run
+        builds_to_run_list = global_build_config_file['builds']
+
+        build_recipes_config_file = None
+        with open(args.buildrecipesconfigfile, "r") as yaml_file:
+            build_recipes_config_file = yaml.safe_load(yaml_file)
+
+        build_farm_hosts_config_file = None
+        with open(args.buildfarmconfigfile, "r") as yaml_file:
+            build_farm_hosts_config_file = yaml.safe_load(yaml_file)
+
+        build_recipes = dict()
+        for section_name, section_dict in build_recipes_config_file.items():
+            if section_name in builds_to_run_list:
+                build_recipes[section_name] = BuildConfig(
+                    section_name,
+                    section_dict,
+                    build_farm_hosts_config_file,
+                    self,
+                    launch_time)
+
+        self.hwdb = RuntimeHWDB(args.hwdbconfigfile)
+
+        self.builds_list = list(build_recipes.values())
+        self.build_ip_set = set()
+
+    def setup(self) -> None:
+        """Setup based on the types of build farm hosts."""
+        for build in self.builds_list:
+            auto_create_bucket(build.s3_bucketname)
+
+        # check to see email notifications can be subscribed
+        get_snsname_arn()
+
+    def request_build_farm_hosts(self) -> None:
+        """Launch an instance for the builds. Exits the program if an IP address is reused."""
+
+        def categorize(seq: List[BuildConfig]) -> Dict[str, List[BuildConfig]]:
+            """Bucket build configs by their build farm host dispatcher name.
+
+            Args:
+                seq: List of build configs to categorize.
+
+            Returns:
+                Dict of lists where the key is the build farm host name and the list holds
```

Comment:
What is a 'build farm host name' and how is it different from and IP address?  I think you should say 'BuildFarmHostDispatcher derived class name'.